### ISSUE TITLE
Crash happening again at CSIOCoreHTTPNetworking.m - Line 230 #13 after updating to v3.10.0

### ISSUE THREAD

**Issue by apcnir** [contentstack/contentstack-ios#19]
Hello I want to report that the issue originally referenced at #13 has started to happen again on version 3.10.0, by taking a look on the latest version of the file [CSIOCoreHTTPNetworking.m](https://github.com/contentstack/contentstack-ios/blob/master/ContentstackInternal/CSIOCoreHTTPNetworking.m) I'm no longer seeing the changes from the PR https://github.com/contentstack/contentstack-ios/pull/14. 

Here's the crash details of version 3.10.0:
```
 Fatal Exception: NSInvalidArgumentException
*** +[NSJSONSerialization dataWithJSONObject:options:error:]: value parameter is nil
```

Stack trace

> Fatal Exception: NSInvalidArgumentException
> 0  CoreFoundation                 0x12586c __exceptionPreprocess
> 1  libobjc.A.dylib                0x6c50 objc_exception_throw
> 2  Foundation                     0x679fc -[NSThread isCancelled]
> 3  Contentstack                   0xc838 -[CSIOCoreHTTPNetworking saveToCacheDataTask:responseObject:] + 230 (CSIOCoreHTTPNetworking.m:230)
> 4  Contentstack                   0xd598 __114-[CSIOCoreHTTPNetworking requestForStack:withURLPath:requestType:params:additionalHeaders:cachePolicy:completion:]_block_invoke + 378 (CSIOCoreHTTPNetworking.m:378)
> 5  Contentstack                   0xe6e0 __59-[CSURLSessionManager dataTaskWithRequest:success:failure:]_block_invoke_2 + 140 (CSURLSessionManager.m:140)
> 6  CFNetwork                      0x1ed60 CFNetServiceBrowserSearchForServices
> 7  CFNetwork                      0x31390 _CFHTTPMessageSetResponseProxyURL
> 8  Foundation                     0x133850 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__
> 9  Foundation                     0x20740 -[NSBlockOperation main]
> 10 Foundation                     0x135ca4 __NSOPERATION_IS_INVOKING_MAIN__
> 11 Foundation                     0x203c8 -[NSOperation start]
> 12 Foundation                     0x13674c __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__
> 13 Foundation                     0x1361d4 __NSOQSchedule_f
> 14 ???                            0x1ac693fb8 (Missing)
> 15 ???                            0x1ac685db0 (Missing)
> 16 ???                            0x1ac68912c (Missing)
> 17 ???                            0x1ac688854 (Missing)
> 18 ???                            0x1ac69693c (Missing)
> 19 ???                            0x1ac697120 (Missing)
> 20 ???                            0x1f85417d8 (Missing)
> 21 ???                            0x1f854876c (Missing)

**Comment by ishaileshmishra** [comment]
Thanks for writing us @apcnir, We are looking into the issue.

**Comment by apcnir** [comment]
Hi @ishaileshmishra I'm writing you back because we tried to make a **downgrade** to the previous version `
v3.9.1` but it's not compiling because of a dependency issue: 
`ContentstackDefinitions.h not found`

This didn't happen in the past before, so it looks like something got mixed on podspecs

**Comment by uttamukkoji** [comment]
Hi @apcnir,

We have resolved the above issue in `v3.10.1`. Closing this issue for now,  If you have any concern or issue you can re-open or create new issue.
